File: backEnd/Systetica_Api/src/main/java/com/frfs/systetica/entity/Agendamento.java
package com.frfs.systetica.entity;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.sun.istack.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

import javax.persistence.*;
import java.io.Serializable;
import java.time.LocalTime;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "agendamento")
@EqualsAndHashCode
public class Agendamento implements Serializable {
    private static final long serialVersionUID = 1L;

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(name = "nome_cliente", length = 300)
    private String nomeCliente;

    @NotNull
    @Column(name = "data_agendamento")
    private String dataAgendamento;

    @NotNull
    @Column(name = "horario_agendamento")
    @JsonFormat(shape = JsonFormat.Shape.STRING, pattern = "HH:mm")
    private LocalTime horarioAgendamento;

    @NotNull
    @Column(name = "data_cadastro")
    private Date dataCadastro;

    @NotNull
    @OneToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "situacao")
    private Situacao situacao;

    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "cliente")
    private Usuario cliente;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "funcionario")
    private Usuario funcionario;

    @NotNull
    @ManyToOne(fetch = FetchType.EAGER)
    @JoinColumn(name = "empresa")
    private Empresa empresa;

    @ManyToMany()
    @JoinTable(
            name = "ass_servico_agendado",
            uniqueConstraints = @UniqueConstraint(columnNames = {"id_agendar_servico", "id_servico"}),
            joinColumns = @JoinColumn(name = "id_agendar_servico"),
            inverseJoinColumns = @JoinColumn(name = "id_servico"))
    private List<Servico> servicos = new ArrayList<>();
}

File: frontEnd/Systetica_App/lib/model/Agendamento.dart
// ignore_for_file: file_names

import 'Servico.dart';
import 'Situacao.dart';
import 'Usuario.dart';

class Agendamento {
  Agendamento({
    this.id,
    this.nomeCliente,
    this.dataAgendamento,
    this.horarioAgendamento,
    this.servicos,
    this.situacao,
    this.cliente,
    this.funcionario,
  });

  int? id;
  String? nomeCliente;
  String? dataAgendamento;
  String? horarioAgendamento;
  List<Servico>? servicos;
  Situacao? situacao;
  Usuario? cliente;
  Usuario? funcionario;

  Agendamento.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    nomeCliente = json['nome_cliente'];
    dataAgendamento = json['data_agendamento'];
    horarioAgendamento = json['horario_agendamento'];
    servicos = json['ass_servico_agendado'] != null
        ? (json['ass_servico_agendado']
            .map<Servico>((e) => Servico.fromJson(e))).toList()
        : [];
    situacao = Situacao.fromJson(json['situacao']);
    cliente =
        json['cliente'] != null ? Usuario.fromJson(json['cliente']) : null;
    funcionario = Usuario.fromJson(json['funcionario']);
  }

  static List<Agendamento> fromJsonList(List json) {
    return json.map((item) => Agendamento.fromJson(item)).toList();
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['id'] = id;
    return _data;
  }
}
